games/static/games/proto.js

```diff
@@ -93,68 +119,147 @@ class Alien{
+                this.squares[i].setGround()
+            }
         }
-        if(alienLocations.includes(i)) {
-            let invader = createInvader(alienLocations[i], width)
-            this.squares[i].setCharacter(invader)
-        } else if (i > width*width-width) {
-            this.squares[i].div.classList.add('ground')
+        let defender = new Defender('defender', this.width)
+        this.squares[this.defenderLocation()].setCharacter(defender)
+    }
+
+    // _movement(direction) {    
+    //     for (let i = 0; i < this.squares.length - 1; i++) {
+    //         this.alienLocations[i] = this.alienLocations[i] + direction
+    //         let nextSquare = this.squares[i + direction]
+    //         if (this.alienLocations.includes(i)) {                
+    //             let invader = createInvader(this.alienLocations[i], this.width)
+    //             nextSquare.setCharacter(invader)            
+    //         }
+    //     }
+    // }
+    // moveRight() {
+    //     this._movement(this.direction)
+    // }
+    // moveLeft() {
+    //     this._movement(this.direction)
+    // }
+    // moveDown() {
+    //     this._movement(this.direction)
+    // }
+    determineDirection() {           
+        // if (this.squares[this.alienLocations[0]].borders.includes('left') && this.direction === -1
+        // || this.squares[this.alienLocations[this.alienLocations.length-1]].borders.includes('right') && this.direction === 1 ) {
+        //     this.direction = this.width
+        // } else if (this.direction === this.width) {
+        //     if (this.squares[this.alienLocations[0]].borders.includes('left')) this.direction = 1
+        //     else this.direction = -1 
+        // }        
+        for (let i = 0; i < this.squares.length - 1; i++) {
+            if (this.squares[i].isRightBorder && this.squares[i].hasAlien && this.direction === 1
+                || this.squares[i].isLeftBorder && this.squares[i].hasAlien && this.direction === -1) {
+                    console.log(this.direction)
+                    this.direction = this.width
+                    console.log(this.direction)
+            } else if (this.direction === this.width) {
+                if (this.squares[i].isRightBorder && this.squares[i].hasAlien) this.direction = 1
+                else this.direction = -1
+            }
         }
     }
+    move() {
+        for (let i = 0; i < this.squares.length - 1; i++) {
+            let square = this.squares[i]
+            if (this.alienLocations.includes(i)) {
+                square.pickCharacterUp() 
+            }
+        }
+        this.determineDirection()
+        for (let i = 0; i < this.squares.length - 1; i++) {
+        this.alienLocations[i] = this.alienLocations[i] + this.direction
+        let nextSquare = this.squares[i + this.direction]
+        if (this.alienLocations.includes(i)) {                
+            let invader = createInvader(this.alienLocations[i], this.width)
```

Comment:
you dont need to create the invader again.
you can get the invader from the sqaure by calling `square.pickCharacterUp()` and put it somewhere else